spec/javascripts/models/beta/story_spec.js

```diff
@@ -246,107 +246,179 @@ describe('Story model', function () {
+
+    const notFeatureTypes = [BUG, CHORE, RELEASE];
+    const estimatedValues = [1, 2, 3, 4];
+
+    describe(`when story type is ${FEATURE}`, () => {
+      notFeatureTypes.forEach(noFeatureType => {
+        describe(`and new story type is ${noFeatureType}`, () => {
+          const story = { _editing: { storyType: FEATURE, estimate: 1 } };
+          const newAttributes = { storyType: noFeatureType }
+          let changedStory;
+          
+          beforeEach(() => {
+            changedStory = Story.editStory(story, newAttributes);
+          })
+
+          it('change story estimate to ""', () => {
+            expect(changedStory._editing.estimate).toEqual('');
+          })
+        
+          it(`change story type is ${noFeatureType}`, () => {
+            expect(changedStory._editing.storyType).toEqual(noFeatureType);
+          })
+        });
       });
-    });
-
-    const notFeatureTypes = ['bug', 'release', 'chore'];
 
-    notFeatureTypes.forEach(type => {
-      it(`change story estimate to null and state to unstarted when storyType is ${type} and sets _isDirty to true`, () => {
-        const story = { _editing: { storyType: 'feature', estimate: '', state: 'unscheduled' } };
-        const newAttributes = { storyType: type };
+      describe(`when state is ${UNSTARTED}`, () => {
+        describe(`when new state is ${UNSCHEDULED}`, () => {
+          it("change estimate to ''", () => {
+            const story = { _editing: { storyType: FEATURE, state: UNSTARTED, estimate: 1 } };
+            const newAttributes = { state: UNSCHEDULED }
+    
+            const changedStory = Story.editStory(story, newAttributes);
+    
+            expect(changedStory._editing.estimate).toEqual('');
+            expect(changedStory._editing.state).toEqual(UNSCHEDULED);
+          });
+        });
+      });
 
-        const changedStory = Story.editStory(story, newAttributes);
+      estimatedValues.forEach(estimatedValue => {
+        describe(`when estimate is ${estimatedValue}`, () => {
+          describe(`when new estimate is ""`, () => {
+            it(`change state to unscheduled `, () => {
+              const story = { _editing: { storyType: FEATURE, estimate: estimatedValue, state: STARTED } };
+              const newAttributes = { estimate: '' };
+  
+              const changedStory = Story.editStory(story, newAttributes);
+  
+              expect(changedStory._editing.state).toEqual(UNSCHEDULED);
+              expect(changedStory._editing.estimate).toEqual('');
+            });
+          });
+        });
+      });
 
-        expect(changedStory).toEqual({
-          _editing: {
-            labels: [],
-            storyType: type,
-            estimate: '',
-            state: 'unstarted',
-            _isDirty: true
-          }
+      describe(`when estimate is ""`, () => {
+        estimatedValues.forEach(estimatedValue => {
+          describe(`when new estimate is ${estimatedValue}`, () => {
```

Comment:
You can apply this style https://github.com/Codeminer42/cm42-central/pull/576/files#diff-e7cad8ffcafac32387e195a4957fe897R257-R273 here.